pkg/controllers/routing/ecmp_vip.go

```diff
@@ -116,7 +116,24 @@ func (nrc *NetworkRoutingController) handleServiceDelete(svc *v1core.Service) {
 		glog.Errorf("Error adding BGP policies: %s", err.Error())
 	}
 
-	nrc.withdrawVIPs(nrc.getAllVIPsForService(svc))
+	activeVIPs, _, err := nrc.getActiveVIPs()
+	if err != nil {
+		glog.Errorf("Failed to get active VIP's on service delete event due to: %s", err.Error())
+		return
+	}
+	activeVIPsMap := make(map[string]string)
+	for _, activeVIP := range activeVIPs {
+		activeVIPsMap[activeVIP] = ""
```

Comment:
Small nitpic here. Since you are only using the map to check to see if a service exists or not later on without always looping over the `activeVIPs` array, you don't actually use the string value here.

As such you could define the map as `make(map[string]bool)` and then the `if` statement becomes a bit more apparent later on with something like, since the when you reference a non-existent value from a map it defaults to a zero value of the type, in this case false:
```
if ! activeVIPsMap[serviceVIP] {
```